src/main/resources/javascript/api.js

```diff
@@ -0,0 +1,249 @@
+/**
+ *
+ * @param httpResponse {Response}
+ * @return {any} the JSON response
+ */
+function readResponse(httpResponse) {
+    if (!httpResponse.ok) {
+        throw new Error('Network response was not ok');
+    }
+    return httpResponse.json();
+}
+
+function readErrors(response) {
+    const errorMessage = response.errors?.map(e => e.message).join(', ');
+    if (errorMessage) {
+        // very likely the user lose their session
+        if (response.errors[0].errorType === "GqlAccessDeniedException") {
+            goToStartPage();
+        } else {
+            throw new Error(errorMessage);
+        }
+    }
+    return response;
+}
+
+function execGraphQL(urlContext, query, variables) {
+    return fetch(urlContext + '/modules/graphql', {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+            query: query,
+            variables: variables
+        })
+    })
+        .then(httpResponse => readResponse(httpResponse))
+        .then(response => readErrors(response));
+}
+
+/**
+ * Create a folder and return a Promise which resolves with the node uuid of the created folder.
+ * @param context {string} the URL context
+ * @param parentPathOrId {string} the parent folder, identified by its path or id, where to create the folder
+ * @param folderName {string} the folder name to create
+ * @return {Promise<string>} a Promise which resolves with the node uuid of the created folder
+ */
+function createFolder(context, parentPathOrId, folderName) {
+    const query = `
```

Comment:
You can add this to get auto-formatting and syntax highlighting

```suggestion
    const query = /* GraphQL */ `
```